specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json

```diff
@@ -5448,88 +5645,766 @@
+      "enum": [
+        "InitialAccess",
+        "Execution",
+        "Persistence",
+        "PrivilegeEscalation",
+        "DefenseEvasion",
+        "CredentialAccess",
+        "Discovery",
+        "LateralMovement",
+        "Collection",
+        "Exfiltration",
+        "CommandAndControl",
+        "Impact",
+        "PreAttack"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "AttackTactic"
+      }
+    },
+    "AutomationRule": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/ResourceWithEtag"
+        }
+      ],
+      "description": "Represents an automation rule.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AutomationRuleProperties",
+          "description": "Automation rule properties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object"
+    },
+    "AutomationRuleAction": {
+      "description": "Describes an automation rule action",
+      "discriminator": "actionType",
+      "properties": {
+        "order": {
+          "description": "The order of execution of the automation rule action",
+          "type": "integer",
+          "format": "int32"
+        },
+        "actionType": {
+          "description": "The type of the automation rule action",
+          "enum": [
+            "ModifyProperties",
+            "RunPlaybook"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "AutomationRuleActionType",
+            "values": [
+              {
+                "description": "Modify an object's properties",
+                "value": "ModifyProperties"
+              },
+              {
+                "description": "Run a playbook on an object",
+                "value": "RunPlaybook"
+              }
+            ]
+          }
+        }
+      },
+      "required": [
+        "order",
+        "actionType"
+      ],
+      "type": "object"
+    },
+    "AutomationRuleCondition": {
+      "description": "Describes an automation rule condition",
+      "discriminator": "conditionType",
+      "properties": {
+        "conditionType": {
+          "description": "The type of the automation rule condition",
+          "enum": [
+            "Property"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "AutomationRuleConditionType",
+            "values": [
+              {
+                "description": "Evaluate an object property value",
+                "value": "Property"
+              }
+            ]
+          }
+        }
+      },
+      "required": [
+        "conditionType"
+      ],
+      "type": "object"
+    },
+    "AutomationRuleRunPlaybookAction": {
+      "description": "Describes an automation rule action to run a playbook",
+      "allOf": [
```

Comment:
Multiple allOf statements is considered a bad practice, it causes problems in languages that don't allow multiple inheritance. See the comment here - https://github.com/Azure/azure-rest-api-specs/pull/12757/files#r585071416
You can add the properties from one of the allOf statements into the schema (for example AutomationRuleRunPlaybookActionProperties).

Reply:
Thanks. When I originally created this PR as draft I had something similar to this (1 allOf and the additional properties defined in the same definition) and got lint check failures saying something along the lines of 'anonymous models are not allowed' and that I had to create a separate definition so that's what I did. I followed the example you referenced and changed it back and the lint check seems to pass now so I guess I used some different syntax originally that passed local model and example validations but failed the linter in the PR.